### PR TITLE
Validation test error models

### PR DESCRIPTION
**Pull request by Gambit142**
I did the following for this milestone

- Added the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of your models' methods and validations.
- Added flash messages in the create actions in all your controllers.
- Made sure that the N+1 problem was solved when fetching all posts and their comments for a user. This was the corresponding wireframe view: [blog all user posts](./images/blog_user_all_posts.png)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
 class PostsController < ApplicationController
   def find_user
-    @user = User.find(params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
   end
 
   def index
     find_user
   end
```
Comment (Lameck1) [inline app/controllers/posts_controller.rb]: - [x] We can improve this logic. Here, we need to find the user first, then find all posts together with their comments. Let's look at what the requirement state:

  - > Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  
  Currently, the logic as implemented does not fetch posts with their associated comments.
  
  Simplify this logic into the index action and do something like this:
  
  ```ruby
  def index
   @user = User.find(params[:user_id])
   @posts = @user.posts.includes(:comments)
  end
  ```

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @Gambit142, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [OPTIONAL] Your requests are failing, consider looking into them. They aren't a concern for now since they are out of scope in this task.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Gambit142) [reply app/controllers/posts_controller.rb]: Thanks for the advice, I have implemented it. The `find_user` helper method does this `User.find(params[:user_id])`, then in the `index` action, I do this `@posts = find_user.posts.includes(:comments)`

**Review by addod19 — APPROVED** [review]
### Hi @Gambit142, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gambit142** [comment]
Thank you for the feedback

**Pull request merged**